# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerLoadInfo.cpp
# Commit: d9efa80409ee
# Full Hash: d9efa80409eed34d06fb989e6b5ebea0f58bd60f
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-11-10 09:24:53
# Regressor Bug: 1657375
# File Overlap Count: 1
# Description:
#   Bug 1657375 - Cache the principal hash value for OffscreenCanvas on worker threads. r=dom-worker-reviewers,asuth,smaug
#   
#   We cannot access ClientWebGLContext::mCanvasElement or its associated
#   nsIPrincipal off the main thread. We use the hash value of the principal
#   to limit how many WebGL contexts a single domain can create. We can
# ==============================================================================

diff -r 2f9159719ec5 -r d9efa80409ee dom/workers/WorkerLoadInfo.cpp
--- a/dom/workers/WorkerLoadInfo.cpp	Tue Nov 09 20:16:04 2021 +0000
+++ b/dom/workers/WorkerLoadInfo.cpp	Tue Nov 09 20:16:05 2021 +0000
@@ -87,6 +87,7 @@
     : mLoadFlags(nsIRequest::LOAD_NORMAL),
       mWindowID(UINT64_MAX),
       mReferrerInfo(new ReferrerInfo(nullptr)),
+      mPrincipalHashValue(0),
       mFromWindow(false),
       mEvalAllowed(false),
       mReportCSPViolations(false),
@@ -156,6 +157,7 @@
     NS_ENSURE_SUCCESS(rv, rv);
   }
 
+  mPrincipalHashValue = aPrincipal->GetHashValue();
   return NS_OK;
 }
 